src/controller/genericmidi.cxx

```diff
@@ -609,19 +609,18 @@ void GenericMIDI::launchScene( int scene )
 		Binding* b = actionToMidi.at(i);
 
 		if ( b->action == GRID_LAUNCH_SCENE ) {
-			for( int i = 0; i < 5; i++ ) {
-				if ( i != scene ) {
-					unsigned char data[3];
-					data[0] = b->status;
-					data[1] = b->data + i;
-					data[2] = 0;
-					writeMidi( data );
-				}
-			}
 			unsigned char data[3];
 			data[0] = b->status;
-			data[1] = b->data + scene;
-			data[2] = 127;
+			data[1] = b->data + i;
+			
+			if ( i != scene ) 
+			{
+				data[2] = 0;
+			} 
+			else 
+			{
+				data[2] = 127;
+			}
```

Comment:
Can improve the above if() else() to this:
`data[2] = (i == scene) * 127;`

Reply:
Thanks for the hint. In which way would this be an improvement? Saving space or also performance?

Reply:
Is not exactly what i red about compiler optimizations, but if this is necessary to get this merged, i will do.